### PR TITLE
Add unit tests for com.baidu.hugegraph.util.StringUtil

### PR DESCRIPTION
**Pull request by Braavos96**
I've analysed your codebase and noticed that com.baidu.hugegraph.util.StringUtil is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

**Comment by javeme** [comment]
@Braavos96 Thank you very much for your code contribution. Yes it's very useful to improve the tests, 
 and look forward to your further contributions.

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Test
+    public void testUnescape() {
+        assertEquals(1, unescape("", "", "").length);
+        assertEquals(1, unescape("foo", "bar", "baz").length);
+    }
+
+    @Test
+    public void testValueOf() {
+        assertEquals(12, valueOf(Integer.class, "12"));
```
Comment (javeme) [inline hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: also add cases for Byte,Short,Long,Float,Double

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.baidu.hugegraph.util;
```
[the remark this answers is not in the document]
Comment (Braavos96) [reply hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: added apache2 license

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.junit.Test;
+import java.util.ArrayList;
+
+import static org.junit.Assert.*;
+import static com.baidu.hugegraph.util.StringUtil.*;
+
+public class StringUtilTest {
```
[the remark this answers is not in the document]
Comment (Braavos96) [reply hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: Refactored class name from StringUtilTest to UnitTestSuite

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    public void testUnescape() {
+        assertEquals(1, unescape("", "", "").length);
+        assertEquals(1, unescape("foo", "bar", "baz").length);
+    }
+
+    @Test
+    public void testValueOf() {
+        assertEquals(12, valueOf(Integer.class, "12"));
+    }
+}
```
[the remark this answers is not in the document]
Comment (Braavos96) [reply hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: file moved

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package com.baidu.hugegraph.util;
+
+import org.junit.Test;
+import java.util.ArrayList;
+
+import static org.junit.Assert.*;
```
[the remark this answers is not in the document]
Comment (Braavos96) [reply hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: class added and import * removed as requested

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import static com.baidu.hugegraph.util.StringUtil.*;
```
[the remark this answers is not in the document]
Comment (Braavos96) [reply hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: class added and import * removed as requested

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+public class StringUtilTest {
+
+    @Test
+    public void testDesc() {
+        assertEquals(".\'\'()", desc("\'\'", new ArrayList<>()));
```
[the remark this answers is not in the document]
Comment (Braavos96) [reply hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: test altered

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java (hunk 1 above)
Comment (Braavos96) [reply hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: additional test cases added

**Comment by Braavos96** [comment]
Hi @javeme, I have amended the test cases to include your comments on it and also signed the CLA.
Apologies I accidentally closed it while pushing the changes :)

File: hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (javeme) [reply hugegraph-core/src/test/java/com/baidu/hugegraph/util/StringUtilTest.java]: I mean that this class should be registered to [UnitTestSuite](https://github.com/hugegraph/hugegraph/blob/master/hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java#L48) in order to be run by mvn test. Please feel free to change the class name back to StringUtilTest.

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/UnitTestSuite.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+ */
+
+package com.baidu.hugegraph.unit.core;
+
+import com.baidu.hugegraph.util.StringUtil;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.ArrayList;
+
+public class UnitTestSuite {
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/UnitTestSuite.java]: I mean that this class should be registered to [UnitTestSuite](https://github.com/hugegraph/hugegraph/blob/master/hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java#L48) in order to be run by mvn test. Please feel free to change the class name back to StringUtilTest.

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/UnitTestSuite.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    @Test
+    public void testDesc() {
+        Assert.assertEquals(".\'\'()", StringUtil.desc("\'\'", new ArrayList<>()));
+        Assert.assertEquals(".1\'2\'3()", StringUtil.desc("1\'2\'3", new ArrayList<>()));
+        Assert.assertEquals(".a\'b\'c()", StringUtil.desc("a\'b\'c", new ArrayList<>()));
+    }
+
+    @Test
+    public void testEscape() {
+        Assert.assertEquals("\u000011213", StringUtil.escape('1', '\u0000', new String[]{"1", "2", "3"}));
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/UnitTestSuite.java]: Please keep the length of one line no more than 80 characters, [click for more details](https://github.com/hugegraph/hugegraph-doc/wiki/HugeGraph%E4%BB%A3%E7%A0%81%E9%A3%8E%E6%A0%BC%E6%8C%87%E5%8D%97)

**Comment by javeme** [comment]
@Braavos96 Thank you very much, there are some minor comments to be addressed.

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/UnitTestSuite.java (hunk 9 above)
Comment (Braavos96) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/UnitTestSuite.java]: I have shortened the assert

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/UnitTestSuite.java (hunk 8 above)
Comment (Braavos96) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/UnitTestSuite.java]: Apologies I have changed the class name back and added  `StringUtilTest.class,` to hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java [hunk 10]
```diff
@@ -46,6 +40,7 @@
     EdgeIdTest.class,
     AnalyzerTest.class,
     JsonUtilTest.class,
+        StringUtilTest.class,
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java]: align with JsonUtilTest

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java [hunk 11]
```diff
@@ -19,18 +19,12 @@
 
 package com.baidu.hugegraph.unit;
 
+import com.baidu.hugegraph.unit.core.*;
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java]: please note don't import "*"

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/StringUtilTest.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+import com.baidu.hugegraph.util.StringUtil;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.ArrayList;
+
+public class StringUtilTest {
+
+    @Test
+    public void testDesc() {
+        Assert.assertEquals(".\'\'()", StringUtil.desc("\'\'", new ArrayList<>()));
```
Comment (Linary) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/StringUtilTest.java]: Please ensure the line don't exceed 80 characters

Comment (zhoney) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/StringUtilTest.java]: align as:
```
        Assert.assertEquals(".1\'2\'3()",
                            StringUtil.desc("1\'2\'3", new ArrayList<>()));
        Assert.assertEquals(".a\'b\'c()",
                            StringUtil.desc("a\'b\'c", new ArrayList<>()));
```

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java (hunk 10 above)
Comment (Braavos96) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java]: Aligned

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java (hunk 11 above)
Comment (Braavos96) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java]: imported each class

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/StringUtilTest.java (hunk 12 above)
Comment (Braavos96) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/StringUtilTest.java]: Changed asserts

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/StringUtilTest.java [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+        Assert.assertEquals(".1\'2\'3()", StringUtil.desc("1\'2\'3", new ArrayList<>()));
+        Assert.assertEquals(".a\'b\'c()", StringUtil.desc("a\'b\'c", new ArrayList<>()));
+    }
+
+    @Test
+    public void testEscape() {
+        Assert.assertEquals(".1\'2\'3()",
+                StringUtil.desc("1\'2\'3", new ArrayList<>()));
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/StringUtilTest.java]: it should be a test for StringUtil.escape()

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java [hunk 14]
```diff
@@ -19,18 +19,19 @@
...
 import com.baidu.hugegraph.unit.core.AnalyzerTest;
 import com.baidu.hugegraph.unit.core.BackendMutationTest;
 import com.baidu.hugegraph.unit.core.CassandraTest;
 import com.baidu.hugegraph.unit.core.ConditionQueryFlattenTest;
 import com.baidu.hugegraph.unit.core.EdgeIdTest;
 import com.baidu.hugegraph.unit.core.JsonUtilTest;
 import com.baidu.hugegraph.unit.core.VersionTest;
+import com.baidu.hugegraph.unit.core.StringUtilTest;
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java]: move to line 22, keep org first

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java [hunk 15]
```diff
@@ -19,18 +19,19 @@
...
+
+import com.baidu.hugegraph.unit.cache.CacheManagerTest;
+import com.baidu.hugegraph.unit.cache.RamCacheTest;
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java]: keep import in alphabetical order

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/StringUtilTest.java (hunk 13 above)
Comment (Braavos96) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/unit/core/StringUtilTest.java]: Apologies, I am not sure how that happened.. but the tests have changed to StringUtil.escape()

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java (hunk 14 above)
Comment (Braavos96) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java]: moved org above other imports

File: hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java (hunk 15 above)
Comment (Braavos96) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java]: Imports are now in alphabetical order

Comment (imbajin) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/unit/UnitTestSuite.java]: > 
> 
> Imports are now in alphabetical order

Maybe u can refer [huge-code-style-props](https://github.com/imbajin/nocode/blob/master/properties/hugegraph-style.xml) or import it if u use `IDEA`, can avoid most of package import, order and alignments issues..

**Pull request merged**